### PR TITLE
[OSDEV-1175] Moderation. Moderation Queue Page connect pagination, sorting and filters (integration)

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-1175](https://opensupplyhub.atlassian.net/browse/OSDEV-1175) Moderation. Moderation Queue Page connect pagination, sorting and filters (integration)

- Integrated getting, paginating, sorting and filtering the moderation events data
- Updated download formats

[OSDEV-1175]: https://opensupplyhub.atlassian.net/browse/OSDEV-1175?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx [hunk 1]
```diff
@@ -118,9 +177,11 @@
                                                     DATE_FORMATS.LONG,
                                                 )}
                                             </TableCell>
-                                            <TableCell>{name}</TableCell>
+                                            <TableCell>
+                                                {cleanedData.name}
+                                            </TableCell>
                                             <TableCell padding="dense">
-                                                {country.name}
+                                                {cleanedData.country.name}
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx:184]: It couldn't be null

File: src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx [hunk 2]
```diff
@@ -118,9 +177,11 @@
                                                     DATE_FORMATS.LONG,
                                                 )}
                                             </TableCell>
-                                            <TableCell>{name}</TableCell>
+                                            <TableCell>
+                                                {cleanedData.name}
+                                            </TableCell>
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx:182]: It couldn't be null

File: src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx [hunk 3]
```diff
@@ -106,7 +165,7 @@ function DashboardModerationQueueListTable({ events, fetching, classes }) {
                                             key={moderationId}
                                             className={classes.rowStyles}
                                             role="button"
-                                            aria-label={`View contribution record for ${name}`}
+                                            aria-label={`View contribution record for ${cleanedData.name}`}
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx:168]: It couldn't be null

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 4]
```diff
@@ -149,11 +243,27 @@
...
-        moderationEvents: { events, fetching, error },
+        moderationEvents: {
+            events,
+            count,
+            page,
+            maxPage,
+            pageSize,
+            fetching,
+            error,
+        },
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx:254]: It couldn't be null

File: src/react/src/reducers/DashboardModerationQueueReducer.js [hunk 5]
```diff
@@ -43,7 +58,45 @@ export default createReducer(
                 moderationEvents: {
                     fetching: { $set: initialState.moderationEvents.fetching },
                     error: { $set: initialState.moderationEvents.error },
-                    events: { $set: data },
+                    events: {
+                        $set: [...state.moderationEvents.events, ...data.data],
+                    },
+                    count: { $set: data.count },
+                },
+            }),
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/reducers/DashboardModerationQueueReducer.js:66]: The logic won't accept it so it fine

File: src/react/src/util/util.moderationEventsXLSX.js [hunk 6]
```diff
@@ -16,14 +16,14 @@ const formatModerationEventsDataForXLSX = moderationEvents =>
     [xlsxHeaders].concat(
         moderationEvents.map(moderationEvent => [
             formatDate(moderationEvent.created_at, DATE_FORMATS.LONG),
-            moderationEvent.name,
-            moderationEvent.country.name,
+            moderationEvent.cleaned_data.name,
+            moderationEvent.cleaned_data.country.name,
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/util/util.moderationEventsXLSX.js:20]: It couldn't be null

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 7]
```diff
@@ -125,19 +213,25 @@
...
             />
         </Paper>
     );
 };
 
 DashboardModerationQueue.defaultProps = {
     events: [],
+    count: 0,
     error: null,
     downloadEventsError: null,
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx]: It's already there.

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 8]
```diff
@@ -125,19 +213,25 @@
             <DashboardModerationQueueListTable
                 fetching={fetching}
                 events={events}
+                count={count}
+                index={page}
+                maxIndex={maxPage}
+                fetchEvents={fetchEvents}
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx:219]: It's used.

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 9]
```diff
@@ -34,26 +47,101 @@
...
+            dispatch(updateAfterDate(date));
+            dispatch(clearModerationEvents());
+            dispatch(
+                updateModerationEventsPage({
+                    page: INITIAL_PAGE_INDEX,
+                    maxPage: INITIAL_PAGE_INDEX,
+                    pageSize,
+                }),
+            );
+            fetchEvents();
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx:124]: We need to call dispatch even if there isn't date.

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 10]
```diff
@@ -34,26 +47,101 @@ const DashboardModerationQueue = ({
...
+            dispatch(updateBeforeDate(date));
+            dispatch(clearModerationEvents());
+            dispatch(
+                updateModerationEventsPage({
+                    page: INITIAL_PAGE_INDEX,
+                    maxPage: INITIAL_PAGE_INDEX,
+                    pageSize,
+                }),
+            );
+            fetchEvents();
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx:144]: We need to call dispatch even if there isn't date.

File: src/react/src/util/propTypes.js [hunk 11]
```diff
@@ -528,11 +514,13 @@ export const moderationEventPropType = shape({
...
     request_type: string,
     source: string,
-    moderation_status: oneOf(['PENDING', 'APPROVED', 'REJECTED']),
+    status: oneOf(['PENDING', 'APPROVED', 'REJECTED']),
     status_change_date: string,
     claim_id: number,
 });
 
+export const moderationEventsPropType = arrayOf(() => moderationEventPropType);
+
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/util/propTypes.js:523]: Done.

File: src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx [hunk 12]
```diff
@@ -86,27 +134,26 @@ function DashboardModerationQueueListTable({ events, fetching, classes }) {
                             <TableRow className={classes.emptyRowStyles} />
                             {events
                                 .slice(
-                                    page * rowsPerPage,
-                                    page * rowsPerPage + rowsPerPage,
+                                    page * pageSize,
+                                    page * pageSize + pageSize,
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueueListTable.jsx:138]: It's not handled automatically on server side

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Overall looks good, left one question and one optimization proposal.

File: src/react/src/util/util.js [hunk 13]
```diff
@@ -260,11 +260,54 @@ export const makeNonStandardFieldsURL = () => '/api/nonstandard-fields/';
...
     `/api/v1/production-locations/${osID}/`;
 
+export const makeGetModerationEventsWithQueryString = (
+    qs,
+    page,
+    pageSize,
+    sortBy,
+    orderBy,
+) =>
+    `/api/v1/moderation-events/?${qs}&sort_by=${sortBy}&order_by=${orderBy}&from=${
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/util/util.js:270]: @VadimKovalenkoSNF Yes, for moderation queue. We decided to use searchAfter field for future OS Hub search implementation.

File: src/react/src/util/util.js [hunk 14]
```diff
@@ -260,11 +260,54 @@ export const makeNonStandardFieldsURL = () => '/api/nonstandard-fields/';
...
+
+    const inputForQueryString = Object.freeze({
+        source: dataSources,
+        status: moderationStatuses,
+        country: countries,
+        date_gte: afterDate,
+        date_lt: beforeDate,
+    });
+
+    return Object.entries(inputForQueryString)
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/util/util.js:295]: @VadimKovalenkoSNF Thank's, Done.

**Review by VadimKovalenkoSNF — APPROVED** [review]
Approved.

**Review by vladsha-dev — APPROVED** [review]
LGTM

**Pull request merged**